IDE/GCC-ARM/Source/tls_client.c

```diff
@@ -0,0 +1,196 @@
+/* tls_client.c
+ *
+ * Copyright (C) 2006-2017 wolfSSL Inc.
+ *
+ * This file is part of wolfSSL.
+ *
+ * wolfSSL is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * wolfSSL is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1335, USA
+ */
+
+
+#include <wolfssl/wolfcrypt/settings.h>
+#include <wolfssl/ssl.h>
+#include <wolfssl/wolfcrypt/logging.h>
+#include <stdio.h>
+
+#define MAXSZ              1024
+
+/*------------------------------------------------------------------------*/
+/* TLS CLIENT */
+/*------------------------------------------------------------------------*/
+static int CbIORecv(WOLFSSL *ssl, char *buf, int sz, void *ctx)
+{
+    int ret = sz;
+
+    (void)ssl;
+    (void)ctx;
+
+    /* TODO: Exchange data over your own transport */
+#if 0
+    ret = usart_read_buffer_wait(&cdc_uart_module, buf, sz);
+    if (ret == STATUS_ERR_TIMEOUT)
+        return WOLFSSL_CBIO_ERR_WANT_READ;
+
+    return (ret == STATUS_OK) ? sz : WOLFSSL_CBIO_ERR_GENERAL;
+#else
+    return ret;
```

Comment:
Shouldn't we have a compiler error  or bsd style function call that would be a compiler error w/o bsd sockets instead of silently failing that will be hard for a new user to debug / understand?

Reply:
Yes I agree and will improve the TLS client example here. Thanks.